imports/database/database-access-object.js

```diff
@@ -0,0 +1,37 @@
+export class DatabaseAccessObject {
+    constructor(collection) {
+        this._collection = collection;
+    }
+
+    find(queryObject) {
+        return this._collection.find(queryObject);
+    }
+
+    findOne(queryObject) {
+        return this._collection.findOne(queryObject);
+    }
+
+    insert(object) {
+        this._collection.schema.validate(object);
+        this._collection.insert(object);
+    }
+
+    update(id, object) {
+        let isValid = true;
+        object.keys().forEach(function (key) {
+            isValid &= this._collection.schema.newContext().validateOne(object, key);
```

Comment:
I'm also a little sketchy on my understanding here, despite reading the docs. I know that unlike schema.validate() which throws an error, context.validate() returns true or false. I think because in this case we're not updating all of the fields in the collection (the base context), we need to use a different context.

I know that this code won't run without the newContext() line.